app/controllers/application_controller.rb

```diff
@@ -105,18 +105,26 @@ def sign_out(*args)
   end
 
   def resolved_authn_context_result
-    @resolved_authn_context_result ||= AuthnContextResolver.new(
-      service_provider: current_sp,
-      vtr: sp_session[:vtr],
-      acr_values: sp_session[:acr_values],
-    ).resolve
+    return @resolved_authn_context_result if defined?(@resolved_authn_context_result)
+
+    if current_sp.nil?
+      @resolved_authn_context_result ||= Vot::Parser::Result.no_sp_result
+    else
+      @resolved_authn_context_result ||= AuthnContextResolver.new(
+        service_provider: current_sp,
+        vtr: sp_session[:vtr],
+        acr_values: sp_session[:acr_values],
+      ).resolve
+    end
   end
 
   def context
     user_session[:context] || UserSessionContext::AUTHENTICATION_CONTEXT
   end
 
   def current_sp
+    return @current_sp if defined?(@current_sp)
+
     @current_sp ||= sp_from_sp_session || sp_from_request_id
```

Comment:
I don't think we need the ||= given the guard above.

Reply:
I'm actually going to get rid of the code here for caching the current SP value. It looks like it causes a lot of issues with controller tests. We can come back to it at some point.